UI/Qt/LocationEdit.cpp

```diff
@@ -37,11 +37,34 @@ LocationEdit::LocationEdit(QWidget* parent)
 
         clearFocus();
 
+        auto query = ak_string_from_qstring(text());
+
+        auto const bang_char = '!'; // should this be a setting?
+
         Optional<StringView> search_engine_url;
-        if (Settings::the()->enable_search())
+        if (Settings::the()->enable_search()) {
             search_engine_url = Settings::the()->search_engine().query_url;
 
-        auto query = ak_string_from_qstring(text());
+            auto const splits = query.split(' ');
+            if (!splits.is_error()) {
+                auto const first = splits.value().first();
+                if (first.starts_with(bang_char)) {
+                    auto exist = WebView::find_search_engine_by_bang(first);
+                    if (exist.has_value()) {
+                        search_engine_url = exist->query_url;
```

Comment:
After some quick testing, other implementations take the first bang and ignore everything else. We should probably return early here.